### ISSUE TITLE
Serializer may have meta class without model

### ISSUE THREAD

**Issue by jihoon-MRX** [tfranzel/drf-spectacular#898]
**Describe the bug**
We have a problem with your new version which is easily fixable, but we report it to let you know that there is an edge case.
In `drf_spectacular/openapi.py` line 675, it checks whether a field contains Meta Class or not. 
However, my team have used custom Serializer subclasses which has Meta class but not model.

**To Reproduce**

Here is the example, 
## Serializer Subclass
```python
class FlatModelSerializer(s.Serializer):
    def create(self, validated_data):
        meta = getattr(self, 'Meta')
        init_func = getattr(meta, 'init_func')
        creation_args = getattr(meta, 'creation_args')
        args = construct_args(validated_data, *creation_args)
        return init_func(*args)
```

## Subclass of FlatModelSerializer
```python
class Example(FlatModelSerializer):
   ...

    class Meta:
        model = ExampleModel
        init_func = ExampleModel.objects.create_action
        creation_args = ('arg1', 'arg2', 'arg3')
```

**Comment by tfranzel** [comment]
Hi @jihoon-MRX, I'm assuming this is due to the newer and stricter handling of `SlugRelatedField`, which previously only defaulted to string.

Could you provide how you use the `SlugRelatedField`. Is there a queryset available? If yes, I just prepared a fix in #899 

Even if we catch that there is no `model` property, there will still be a warning though.

**Comment by jihoon-MRX** [comment]
Yes they are querysets available. Thanks for the update!
